packages/block-library/src/template-part/edit/revert-tool.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { cleanForSlug } from '@wordpress/url';
+import { ToolbarButton } from '@wordpress/components';
+
+export default function RevertToDefault( {
+	postId,
+	attributePostId,
+	setAttributes,
+} ) {
+	const defaultTemplatePart = useSelect(
+		( select ) => {
+			const {
+				meta: { original_slug: originalSlug, theme },
+			} = select( 'core' ).getEntityRecord(
+				'postType',
+				'wp_template_part',
+				postId
+			);
+
+			// For a theme supplied template part we would expect original_slug to be
+			// present in the meta given the field we added in template-loader.
+			// But inspecting the theme supplied template part or customized/published version
+			// 'theme' seems to be the only meta value present.
```

Comment:
This is where I am currently stuck.  As outlined in the comments, it seems like the `original_slug` we added in the meta of the auto-draft in template-loader isn't actually being returned in the meta here. 🤔